### ISSUE TITLE
Cypress assertions pass silently where they are expected to fail

### ISSUE THREAD

**Issue by ranjithnori** [cypress-io/cypress#9223]
### Context

We are trying to mock some API requests which are common across all pages of our app. 
We wrote a custom command for this and ran it inside `beforeEach` in our tests. 

### Current behavior

We found some inconsistent behavior when we run some of our tests. We can only reproduce it once in every 7-10 test runs. 

1. Sometimes the custom command called in `beforeEach` doesn't run even though `beforeEach` runs.
2. Some of our assertions which use `cy.get` chained with `cy.contains` pass silently although no elements are matched. 
3. `beforeEach` doesn't appear in cypress command logs although its executed.

Note: We kept some `console.log`s and confirmed this behavior

Here's a screenshot when this behavior occurred:

![image](https://user-images.githubusercontent.com/4106656/99353895-71580c80-28cb-11eb-981a-d90786d05b48.png)

A screenshot where the same test passed as expected:

![image](https://user-images.githubusercontent.com/4106656/99359982-e16b9000-28d5-11eb-83c0-6dcd8ddc3926.png)

### Desired behavior

1. Custom commands must run consistently in `beforeEach`
2. The first assertion with 0 matched elements must fail consistently (please refer to the screenshot).
3. When `beforeEach` runs, it must be displayed in the command logs consistently

### Additional information

We used `cy.route2` for stubbing API calls and started noticing these issues after enabling `experimentalNetworkStubbing`. 
This inconsistency is reproducible in our local environment and Github CI

### Test code to reproduce

We're still trying to narrow down to the root cause of these problems. So, any pointers on what might be causing this inconsistent behavior would be great until we can give a repo that you can clone and run. 

### Versions

Cypress version: Can reproduce with 5.5.0 and 5.6.0
Browser version: Chrome 85.0.4183.102 (Official Build) (64-bit)

**Comment by jennifer-shehane** [comment]
Can you provide a reproducible example? I won't be able to guess what is causing this issue without seeing the code in question. Here are some tips for providing a [Short, Self Contained, Correct, Example](http://sscce.org/) and our own [Troubleshooting Cypress](https://on.cypress.io/troubleshooting) guide.

**Comment by ranjithnori** [comment]
[Here](https://github.com/ranjithnori/cypress-test-tiny) is a repo which contains two minimal reproducible cases.

spec 1:

![Screenshot from 2020-11-18 17-11-01](https://user-images.githubusercontent.com/4106656/99526710-aa72a880-29c1-11eb-8bff-a11291ccf490.png)

spec 2:

All the test commands are skipped in test 2 and it just passes. The only difference from `spec1` is we removed `beforeEach` from `Block 2`

![Screenshot from 2020-11-18 17-11-56](https://user-images.githubusercontent.com/4106656/99526726-b0688980-29c1-11eb-8c9f-c87aa443dfdd.png)

The issue can not be replicated when any of the following lines are removed:
```
window.localStorage.setItem('secret', 'random');
cy.route2('https://staging.api.kisi.io/user').as('getBaseUser');
```

**Comment by mmacaula** [comment]
I'm also seeing this behavior.  

For me, the left side panel lists either my `cy.get(selector)` or `cy.contains(text)`  with a `0` on the right.  I believe it should wait for those to find something by the timeout by cypress seems to be just skipping over them.  

I'm relying the default assertion if that helps.  I'm not appending anything like `should('have.length',1)` or anything. 

Some anecdotal info:  it seems to work sometimes.  sometimes not.  If I keep refreshing sometimes it makes it through.  I've been using Chrome but when I switched to electron I haven't seen this issue.

**Comment by ranjithnori** [comment]
@jennifer-shehane Update: The issues can be reproduced even after upgrading to 6.0.0. Here's the [repo link](https://github.com/ranjithnori/cypress-test-tiny) for your reference

**Comment by ranjithnori** [comment]
Hi @jennifer-shehane, did you get a chance to look at the reproducible test cases ?

**Comment by jennifer-shehane** [comment]
@ranjithnori This is fixed in 6.1.0. Please update to the [current version of Cypress](https://on.cypress.io/changelog).
